test/functional/mirror/mirror-set-unset.bats

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env bats
+
+# Author: Otavio Pontes
+# Email: otavio.pontes@intel.com
+
+load "../testlib"
+
+global_setup() {
+
+	create_test_environment "$TEST_NAME"
+
+}
+
+test_setup() {
+
+	# do nothing
+	return
+
+}
+
+test_teardown() {
+
+	sudo rm -rf "$TARGETDIR"/etc/swupd
+
+}
+
+global_teardown() {
+
+	destroy_test_environment "$TEST_NAME"
+
+}
+
+@test "MIR011: Set/Unset a full mirror" {
+
+	run sudo sh -c "$SWUPD mirror --set https://example.com/swupd-file $SWUPD_OPTS"
+	assert_status_is 0
+	expected_output=$(cat <<-EOM
+		Mirror url set
+		Distribution:      Swupd Test Distro
+		Installed version: 10
+		Version URL:       https://example.com/swupd-file
+		Content URL:       https://example.com/swupd-file
+	EOM
+	)
+	assert_is_output "$expected_output"
+	assert_equal "https://example.com/swupd-file" "$(<"$TARGETDIR"/etc/swupd/mirror_contenturl)"
+	assert_equal "https://example.com/swupd-file" "$(<"$TARGETDIR"/etc/swupd/mirror_versionurl)"
+
+	run sudo sh -c "$SWUPD mirror --unset $SWUPD_OPTS"
+	assert_status_is 0
+
+	expected_output=$(cat <<-EOM
+		Mirror url configuration removed
+		Distribution:      Swupd Test Distro
+		Installed version: 10
+		Version URL:       .*web-dir
+		Content URL:       .*web-dir
+	EOM
+	)
+	assert_regex_is_output "$expected_output"
```

Comment:
just as FYI, instead of using:
`Content URL:       .*web-dir` with `assert_regex_is_output "$expected_output"`

you could use:
`Content URL:       $WEBDIR` with `assert_is_output "$expected_output"`

I generally try to avoid regex as possible when comparing the output, it makes it harder to find errors when the test fails, because the line with regex characters is always gonna show up in the diff.

Reply:
$WEBDIR is pointinting to web-dir directory but I need the full url. Is there any variable exporting that value? The url is something like:

    file:///home/otaviobp/swupd-client/mirror-set-unset_1/web-dir

Reply:
ohh, in that case I think you can use `$TEST_DIRNAME/web-dir`